mlrun/launcher/client.py

```diff
@@ -60,6 +61,16 @@ def prepare_image_for_deploy(runtime: "mlrun.runtimes.BaseRuntime"):
         ):
             image = mlrun.mlconf.function_defaults.image_by_kind.to_dict()[runtime.kind]
 
+        # Warn if user explicitly set the deprecated mlrun/ml-base image
+        if image and "mlrun/ml-base" in image:
+            warnings.warn(
+                "'mlrun/ml-base' image is deprecated in 1.10.0 and will be removed in 1.12.0, "
+                "use 'mlrun/mlrun' instead. "
+                "For clients >= 1.10.0, the image is automatically replaced with 'mlrun/mlrun'. ",
+                # TODO: Remove this in 1.12.0
```

Comment:
you can see what's the client by checking the current version and mention the image would be automatically replaced